lib/chef/provider/machine.rb

```diff
@@ -19,12 +19,12 @@ def whyrun_supported?
   end
 
   action :allocate do
-    if current_driver && current_driver.driver_url != new_driver.driver_url
-      raise "Cannot move '#{machine_spec.name}' from #{current_driver.driver_url} to #{new_driver.driver_url}: machine moving is not supported.  Destroy and recreate."
-    end
     if !new_driver
       raise "Driver not specified for machine #{machine_spec.name}"
     end
+    if current_driver && current_driver.driver_url != new_driver.driver_url
+      raise "Cannot move '#{machine_spec.name}' from #{current_driver.driver_url} to #{new_driver.driver_url}: machine moving is not supported.  Destroy and recreate."
+    end
     new_driver.allocate_machine(action_handler, machine_spec, new_machine_options)
```

Comment:
This change means that if you remove your driver settings (and you are only using environmental variables) you will get the "Driver not specified for machine" message instead of a method being called on a nil object.